types/extrinsic_status.go

```diff
@@ -0,0 +1,186 @@
+// Go Substrate RPC Client (GSRPC) provides APIs and types around Polkadot and any Substrate-based chain RPC calls
+//
+// Copyright 2019 Centrifuge GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/centrifuge/go-substrate-rpc-client/scale"
+)
+
+// ExtrinsicStatus is an enum containing the result of an extrinsic submission
+type ExtrinsicStatus struct {
```

Comment:
This extrinsicStatus is interesting, is this obj returned by the SubmitAndWatch logic only right? Might be good to wrap here the manual parsing of the status I setup in the go-cent project eventually.

Reply:
Unfortunately yes, it is only returned by the submit and watch logic, and we can't access it via another method. That's why I think mid-term, we might want to subscribe and only manually verify the extrinsic has been included in a block if we run into a timeout.